rdrf/rdrf/admin.py

```diff
@@ -129,93 +161,133 @@ def export_registry(registry, request):
-        if not request.user.is_superuser:
-            user = get_user_model().objects.get(username=request.user)
-            return Registry.objects.filter(registry__in=[reg.id for reg in user.registry.all()])
+    enable_registration_action.short_description = _("Enable registration")
 
-        return Registry.objects.all()
+    def disable_registration_action(self, request, registry_models_selected):
+        unchanged = []
 
-    def has_add_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+        for registry in registry_models_selected:
+            if not registry.has_feature(RegistryFeatures.REGISTRATION):
+                unchanged.append(str(registry))
+            else:
+                registry.remove_feature(RegistryFeatures.REGISTRATION)
+                registry.save()
 
-    def has_change_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+                messages.success(request, _(f"Registration disabled for {registry}"))
 
-    def has_delete_permission(self, request, *args, **kwargs):
-        if request.user.is_superuser:
-            return True
-        return False
+        if unchanged:
+            messages.info(request, _(f"'{', '.join(desc for desc in unchanged)}' already had registration disabled"))
 
-    def get_urls(self):
-        original_urls = super(RegistryAdmin, self).get_urls()
+    disable_registration_action.short_description = _("Disable registration")
 
-        return original_urls
+    def create_notifications_action(self, request, registry_models_selected):
+        for registry in registry_models_selected:
+            # TODO: Add default notifications for all registration types. Needs different recipients.
+            try:
+                with transaction.atomic():
+                    notification = EmailNotification(
+                        description=EventType.NEW_PATIENT,
+                        registry=registry,
+                        recipient="{{ patient.user.email }}",
+                    )
+                    notification.save()
 
-    def get_readonly_fields(self, request, obj=None):
-        "Registry code is readonly after creation"
-        return () if obj is None else ("code",)
+                    notification.email_templates.set(
+                        EmailTemplate.objects.filter(default_for_notification=EventType.NEW_PATIENT)
+                    )
+                    notification.save()
+
+                    messages.success(request, render_to_string("admin/notifications_added.html", {
+                        "registry": registry,
+                    }))
+            except IntegrityError as e:
+                logger.error("Failed to create default registration notifications: %s", e)
+                messages.error(request, _("Failed to create default registration notifications"))
```

Comment:
This error message should include the registry in it, so we know on which registry it failed.